File: examples/hello_world/python/hello_world.py
"""
SPDX-FileCopyrightText: Copyright (c) 2023-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
SPDX-License-Identifier: Apache-2.0
"""  # noqa: E501

from holoscan.conditions import CountCondition
from holoscan.core import Application, Operator, OperatorSpec

# define custom Operators for use in the demo


class HelloWorldOp(Operator):
    """Simple hello world operator.

    This operator has no ports.

    On each tick, this operator prints out hello world messages.
    """

    def setup(self, spec: OperatorSpec):
        pass

    def compute(self, op_input, op_output, context):
        print("")
        print("Hello World!")
        print("")


# Now define a simple application using the operator defined above


class HelloWorldApp(Application):
    def compose(self):
        # Define the operators
        hello = HelloWorldOp(self, CountCondition(self, 1), name="hello")

        # Define the one-operator workflow
        self.add_operator(hello)


def main():
    app = HelloWorldApp()
    app.run()


if __name__ == "__main__":
    main()

File: examples/hello_world/cpp/hello_world.cpp
/*
 * SPDX-FileCopyrightText: Copyright (c) 2022-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

#include <holoscan/holoscan.hpp>
#include <iostream>

namespace holoscan::ops {

class HelloWorldOp : public Operator {
 public:
  HOLOSCAN_OPERATOR_FORWARD_ARGS(HelloWorldOp)

  HelloWorldOp() = default;

  void setup(OperatorSpec& spec) override {}

  void compute([[maybe_unused]] InputContext& op_input, [[maybe_unused]] OutputContext& op_output,
               [[maybe_unused]] ExecutionContext& context) override {
    std::cout << '\n';
    std::cout << "Hello World!\n";
    std::cout << '\n';
  }
};

}  // namespace holoscan::ops

class HelloWorldApp : public holoscan::Application {
 public:
  void compose() override {
    using namespace holoscan;

    // Define the operators
    auto hello = make_operator<ops::HelloWorldOp>("hello", make_condition<CountCondition>(1));

    // Define the one-operator workflow
    add_operator(hello);
  }
};

int main() {
  auto app = holoscan::make_application<HelloWorldApp>();
  app->run();

  return 0;
}
